File: 2024/day22/part1.py
def next_secret(n):
    n = ((n << 6) ^ n) & ((1 << 24) - 1)
    n = (n >> 5) ^ n
    n = ((n << 11) ^ n) & ((1 << 24) - 1)

    return n


with open("input.txt") as f:

    sum = 0
    for secret_number in f:
        secret_number = int(secret_number[:-1])

        for _ in range(2000):
            secret_number = next_secret(secret_number)
        print(secret_number)

        sum += secret_number
    print(f"sum = {sum}")

File: 2025/day05/part1.ts
const input = await Deno.readTextFile("input.txt");

const [rangesStr, productsStr] = input.trim().split("\n\n");

const ranges = rangesStr.split("\n").map(function (line): [number, number] {
    return line.split("-").map(Number) as [number, number];
});

const products = productsStr.split("\n").map(Number);

// console.log(ranges);
// console.log(products);

let count = 0;
for (const product of products) {
    let valid = false;
    for (const range of ranges) {
        if (product >= range[0] && product <= range[1]) {
            valid = true;
        }
    }

    if (valid) {
        // console.log(`${product}: valid`);
        count++;
    }
}

console.log(count);
